### ISSUE TITLE
no-sequences rule in arrow function body returning object is not aplicable

### ISSUE THREAD

**Issue by langpavel** [eslint/eslint#6082]
**In short:**
It is not possible to apply `no-sequences` rule in return part of arrow function in form
*\<argumentList>* **=>** *\<expression>* where *\<expression>* is *\<objectLiteral>* because this object literal *must* be enclosed in parentheses.

**Example:**
I have this *bad* piece of code and want `no-sequences` rule to warn me about:
```js
export default injectIntl(connect(state => ({
  auth: state.auth,
}, { // <-- BAD
  login,
}))(LoginForm));
```
`no-sequences` rule is not designed for this situation.
Correct code should be:
```js
export default injectIntl(connect(state => ({
  auth: state.auth,
}), {
  login,
})(LoginForm));
```
Sadly, both samples executes and both without lint warnings.
Rule `no-sequences` is not applied if comma expression is explicitly enclosed in parentheses.
But it is also impossible to write arrow function returning object without enclosing it into parenthese so no-sequences rule is not usable in this case.

**Proposed Solution**

If there is arrow (e.g. `(arg) => ({result: arg})`), `no-sequences` rule can be enabled in result expression (this is `{result: arg}` inside parentheses)
So it can report if it spot `(arg) => ({result: arg}, {foo: 123})`

The rule is not fired because of this on line **95** of the `no-sequences` rule:
```js
if (isParenthesised(node)) {
    return;
}
```
Should be changed to something similar to:
```js
if (isParenthesised(node) && !isArrowFunctionReturnExpression(node)) {
    return;
}
```

Thanks to @pedrottimark and @vitorbal for help on Gitter.

**Comment by vitorbal** [comment]
FWIW, here's another simplified example where I would expect a warning, but don't get one:
```javascript
state => ({ foo: 'foo' }, { bar: 'bar' })
```

**Comment by langpavel** [comment]
@vitorbal Thanks, I don't write this? lol :-)

**Comment by vitorbal** [comment]
ah, you did :) missed it in the Proposed Solution section, sorry 😄

**Comment by mysticatea** [comment]
Thank you for this issue.

I confirmed it on the online demo.
![image](https://cloud.githubusercontent.com/assets/1937871/15017037/efa86e7e-124e-11e6-87c7-43241ac68353.png)

I'll fix it.

**Comment by platinumazure** [comment]
Oops, I selected "enhancement" when I meant "evaluating". Sorry about that. Thanks @mysticatea for confirming and getting the labels right :smile:

**Comment by mysticatea** [comment]
@platinumazure labels were added at the same time completely :smile:

**Comment by langpavel** [comment]
I'm expecting this behavior:
```js
state => ({ foo: 'foo' },  { bar: 'bar' }); // bad
state => (({ foo: 'foo' },  { bar: 'bar' })); // good
state => (bar,  foo); // good
// state => { foo: 'foo' }; // syntax error
```

**Comment by mysticatea** [comment]
```js
state => ({ foo: 'foo' },  { bar: 'bar' }); // bad
state => (({ foo: 'foo' },  { bar: 'bar' })); // good
state => (bar,  foo); // bad
state => ((bar,  foo)); // good

state => bar, foo // this is (state => bar), foo
```

**Comment by langpavel** [comment]
Ok @mysticatea makes sense :+1:

**Comment by langpavel** [comment]
You are blazingly fast! Thanks!
